### PR TITLE
fix(caldav): add EXDATE and EXRULE to confidential object

### LINKED ISSUE #5860: Removed Single Occurrence of Reoccurring, Confidential Event Remains on Shared Calendar
### Steps to reproduce

1. Create a reoccurring event that only shows as "busy" when shared.
2. Share this calendar.
3. Delete an occurrence of this series.

### Expected behavior

The single occurrence is deleted from both the owner calendar and the shared, confidential version of the calendar.

### Actual behaviour

Occurrence is deleted from the owner calendar, but it remains on the shared calendar (but is not listed in the owner trash; I assume it's actually being made invisible).

Changing the reoccurring event to show "full" details when shared, caused the deleted event to disappear from the shared calendar as well.

### Calendar app version

4.6.6

### CalDAV-clients used

DAVx5

### Browser

Brave 1.63.174

### Client operating system

Windows 11

### Server operating system

Raspbian Bookworm

### Web server

Apache

### Database engine version

PostgreSQL

### PHP engine version

PHP 8.1

### Nextcloud version

28.0.3

### Updated from an older installed version or fresh install

Fresh install

### List of activated apps

_No response_

### Nextcloud configuration

_No response_

### Web server error log

_No response_

### Log file

_No response_

### Browser log

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by miaulalala**
* Resolves: https://github.com/nextcloud/calendar/issues/5860

## Summary

The EXDATE / EXRULE properties are lost when a confidential object is created.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by kesselb — APPROVED** [review]

**Review by tcitworld — APPROVED** [review]
Makes total sense

File: apps/dav/lib/CalDAV/CalendarObject.php [hunk 1]
```diff
@@ -119,6 +119,8 @@ private function createConfidentialObject(Component\VCalendar $vObject) {
 					case 'DURATION':
 					case 'DTEND':
 					case 'CLASS':
+					case 'EXRULE':
+					case 'EXDATE':
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/CalendarObject.php:123]: I think `RDATE` might need to be added and possibly `RECURRENCE-ID` as well.

**Pull request merged**